lib/mix/test/mix/tasks/deps_test.exs

```diff
@@ -193,6 +193,20 @@ defmodule Mix.Tasks.DepsTest do
 
   ## deps.unlock
 
+  test "cleans and recompiles artifacts if --force given" do
+    Mix.Project.push SuccessfulDepsApp
+
+    in_fixture "deps_status", fn ->
+      Mix.Tasks.Deps.Compile.run []
+      File.touch! "deps/ok/clean-me"
```

Comment:
We don't want to clean the deps/ok directory. We want to clean only the stuff inside _build like _build/dev/lib/ok. If you clean deps/ok, we will no longer have the source and be unable to compile the dependency.